io.c

```diff
@@ -1138,6 +1144,12 @@ rb_io_wait_readable(int f)
 int
 rb_io_wait_writable(int f)
 {
+    VALUE selector = rb_current_thread_selector();
+    if (selector != Qnil) {
+	VALUE result = rb_funcall(selector, rb_intern("wait_readable"), 1, INT2NUM(f));
```

Comment:
did you mean `"wait_writable"` ?

Reply:
Yes, sorry, my bad. I will revisit this PR soon, I have been working on https://bugs.ruby-lang.org/issues/14739

Thanks so much for taking time to review this PR. It's just a proof of concept. I will spend more time on it.